local/gugcat/index.php

```diff
@@ -60,12 +60,17 @@
     $groups = groups_get_all_groups($course->id, $userid=0, $groupingid, $fields='g.*');
 
     $scaleid = $selectedmodule->gradeitem->scaleid;
+    $gradetype = $selectedmodule->gradeitem->gradetype;
+    $grademax = $selectedmodule->gradeitem->grademax;
+
     if (is_null($scaleid)){
-        $gradetype = $selectedmodule->gradeitem->gradetype;
-        $grademax = $selectedmodule->gradeitem->grademax;
         $valid_22point_scale = local_gugcat::is_grademax22($gradetype, $grademax);
         $scaleid = $valid_22point_scale ? null : $scaleid;
     }
+    else{
+        $valid_22point_scale = local_gugcat::is_scheduleAscale($gradetype, $grademax);
+        $scaleid = $valid_22point_scale ? null : $scaleid;
+    }
```

Comment:
Assigning null to scaleid when it is valid 22 point scale may cause some issue if the grade type is scale. I think we can replace this with below line. Will just approve for now and to be updated later.

$valid_22point_scale = is_null($scaleid) ? local_gugcat::is_grademax22($gradetype, $grademax) : local_gugcat::is_scheduleAscale($gradetype, $grademax);